### PR TITLE
Added types-aiobotocore

### PR DESCRIPTION
**Pull request by peterbygrave**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/types-aiobotocore/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  skip: true  # [py<38]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
```
Comment (peterbygrave) [inline recipes/types-aiobotocore/meta.yaml:20]:
```suggestion
    - python
    - pip
    - setuptools
```

File: recipes/types-aiobotocore/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  run:
+    - python
+    - botocore-stubs
+    - typing-extensions >=4.1.0  # [py<312]
+
+test:
+  imports:
+    - aiobotocore-stubs
+  commands:
+    - pip check
```
Comment (peterbygrave) [inline recipes/types-aiobotocore/meta.yaml:31]:
```suggestion
  commands:
    - pip check
    - test -f $SP_DIR/aiobotocore-stubs/__init__.pyi  # [unix]
    - if not exist %SP_DIR%\\aiobotocore-stubs\\__init__.pyi exit 1  # [win]
```

File: recipes/types-aiobotocore/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - pip
+    - setuptools
+  run:
+    - python
+    - botocore-stubs
+    - typing-extensions >=4.1.0  # [py<312]
+
+test:
+  imports:
+    - aiobotocore-stubs
```
Comment (peterbygrave) [inline recipes/types-aiobotocore/meta.yaml:29]:
```suggestion
```

**Comment by peterbygrave** [comment]
@conda-forge/staged-recipes @conda-forge/help-python Ready for review, thank you!

**Comment by peterbygrave** [comment]
@conda-forge/help-python Ready for review, thank you!

File: recipes/types-aiobotocore/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+    - setuptools
+  run:
+    - python
+    - botocore-stubs
+    - typing-extensions >=4.1.0  # [py<312]
```
Comment (ocefpaf) [inline recipes/types-aiobotocore/meta.yaml:25]: We can make this noarch. It should be OK to install typing-extensions on py312.

```suggestion
    - typing-extensions >=4.1.0
```

Comment (peterbygrave) [reply recipes/types-aiobotocore/meta.yaml]: Thanks for suggestion. I just copied verbatim from grayskull 🤣

**Pull request merged**